crates/turbopack-ecmascript/src/chunk/optimize.rs

```diff
@@ -267,6 +269,39 @@ async fn merge_duplicated_and_contained(
     Ok(())
 }
 
+async fn merge_by_size(chunks: &[EcmascriptChunkVc]) -> Result<Vec<EcmascriptChunkVc>> {
+    let mut merged = Vec::new();
+    let mut current = Vec::new();
+    let mut current_items = 0;
+    for &chunk in chunks {
+        let chunk_items = *chunk.chunk_items_count().await?;
+        if chunk_items >= MAX_CHUNK_ITEMS_PER_CHUNK {
+            // chunk is too big, keep it separate
+            merged.push(chunk);
```

Comment:
Not sure if ordering is important, but this places the new chunk before whatever already in `current`.

Reply:
It's not important as long it's deterministic